pulp_docker/app/views.py

```diff
@@ -14,3 +57,235 @@ def get(self, request):
+        port = redirect['PORT'] or request.get_port()
+        prefix = redirect['PATH_PREFIX']
+
+        if not enabled:
+            return HttpResponseNotFound()
+
+        path = os.path.join(
+            prefix.strip('/'),
+            self._published_path(request)
+        )
+
+        url = '{scheme}://{host}:{port}/{path}'.format(
+            scheme=request.scheme,
+            host=host,
+            path=path,
+            port=port)
+
+        log.debug(_('Redirected: %(u)s'), {'u': url})
+        response = HttpResponseRedirect(url)
+        return response
+
+    def _dispatch(self, path, headers):
+        """
+        Dispatch to the appropriate responder (method).
+
+        Args:
+            path (str): The fully qualified path to the file to be served.
+
+        Returns:
+            django.http.StreamingHttpResponse: on found.
+            django.http.HttpResponseNotFound: on not-found.
+            django.http.HttpResponseForbidden: on forbidden.
+            django.http.HttpResponseRedirect: on redirect to the streamer.
+        """
+        server = settings.CONTENT['WEB_SERVER']
+
+        try:
+            responder = self.RESPONDER[server]
+        except KeyError:
+            raise ValueError(_('Web server "{t}" not supported.').format(t=server))
+        else:
+            return responder(self, path, headers)
+
+    # Mapping of responder-method based on the type of web server.
+    RESPONDER = {
+        'django': _django,
+        'apache': _apache,
+        'nginx': _nginx,
+    }
+
+
+class TagView(views.APIView, ServeContentMixin):
+    """
+    APIView for Docker Registry v2 tag API
+    """
+
+    authentication_classes = []
+    permission_classes = []
+    content_negotiation_class = IgnoreClientContentNegotiation
+
+    def get(self, request, path, tag_name):
+        """
+        Return a response to the "GET" action.
+        """
+        distribution = get_object_or_404(DockerDistribution, base_path=path)
+        try:
+            ca = ContentArtifact.objects.get(
+                content__in=distribution.publication.repository_version.content,
+                relative_path=tag_name)
+            content = ca.content.cast()
+            if content.manifest:
+                headers = {'Content-Type': content.manifest.media_type}
+            else:
+                headers = {'Content-Type': content.manifest_list.media_type}
+
+        except MultipleObjectsReturned:
+            log.debug(
+                _('Multiple (pass-through) matches for {b}/{p}'),
+                {
+                    'b': distribution.base_path,
+                    'p': tag_name,
+                }
+            )
+
+        except ObjectDoesNotExist:
+            pass
+        else:
+            artifact = ca.artifact
+            if artifact:
+                return self._dispatch(artifact.file.name, headers)
+            else:
+                raise ArtifactNotFound(path)
+
+        raise PathNotResolved(path)
```

Comment:
This only occurs for an ObjectDoesNotExist exception right? If so, I think this would be more clear if this is raised on line 221 instead of a pass. Similar for other views.

Reply:
I'd like to keep it this way for now. I am open to revisit this in another PR.